services/learn-card-network/brain-service/src/helpers/connection.helpers.ts

```diff
@@ -1,156 +1,162 @@
-            throw new TRPCError({
-                code: 'CONFLICT',
-                message: 'Profiles are already connected!',
-            });
-        }
+  if (validate) {
+    if (await areProfilesConnected(source, target)) {
+      throw new TRPCError({
+        code: 'CONFLICT',
+        message: 'Profiles are already connected!',
+      });
+    }
 
-        const pendingRequestFromTarget =
-            (
-                await target.findRelationships({
-                    alias: 'connectionRequested',
-                    where: { relationship: {}, target: { profileId: source.profileId } },
-                })
-            ).length > 0;
+    const pendingRequestFromTarget =
+      (
+        await target.findRelationships({
+          alias: 'connectionRequested',
+          where: { relationship: {}, target: { profileId: source.profileId } },
+        })
+      ).length > 0;
 
-        if (!pendingRequestFromTarget) {
-            throw new TRPCError({
-                code: 'NOT_FOUND',
-                message:
-                    'No connection request found. Please try sending them a connection request!',
-            });
-        }
+    if (!pendingRequestFromTarget) {
+      throw new TRPCError({
+        code: 'NOT_FOUND',
+        message: 'No connection request found. Please try sending them a connection request!',
+      });
     }
+  }
+
+  await Promise.all([
+    Profile.deleteRelationships({
+      alias: 'connectionRequested',
+      where: {
+        source: { profileId: source.profileId },
+        target: { profileId: target.profileId },
+      },
+    }),
+    Profile.deleteRelationships({
+      alias: 'connectionRequested',
+      where: {
+        source: { profileId: target.profileId },
+        target: { profileId: source.profileId },
+      },
+    }),
+    source.relateTo({ alias: 'connectedWith', where: { profileId: target.profileId } }),
+    target.relateTo({ alias: 'connectedWith', where: { profileId: source.profileId } }),
+  ]);
 
-    await Promise.all([
-        Profile.deleteRelationships({
-            alias: 'connectionRequested',
-            where: {
-                source: { profileId: source.profileId },
-                target: { profileId: target.profileId },
-            },
-        }),
-        Profile.deleteRelationships({
-            alias: 'connectionRequested',
-            where: {
-                source: { profileId: target.profileId },
-                target: { profileId: source.profileId },
-            },
-        }),
-        source.relateTo({ alias: 'connectedWith', where: { profileId: target.profileId } }),
-        target.relateTo({ alias: 'connectedWith', where: { profileId: source.profileId } }),
-    ]);
+  await SendPushNotification({
+    to: source.profileId,
+    message: `${target.displayName} has accepted your connection request!`,
+    title: 'Connection Accepted',
+    actionType: 'redirect',
+    url: '/contacts',
```

Comment:
This is okay for now - ideally in the future we might want to refactor this to be a custom uri, like data: `lcn:redirect:JSON.stringify({ url: '/contacts' })` or something like that